clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -767,6 +767,11 @@ public void assign(List<TopicPartition> partitions) {
 
         // init any new fetches (won't resend pending fetches)
         Cluster cluster = this.metadata.fetch();
+        Map records = fetcher.fetchedRecords();
```

Comment:
Also it'd be nice to have a comment basically saying that this check is needed since fetches may have already returned in another API call (e.g. commit).